### DISCUSSION TITLE
How to enforce Join Precedence

### DISCUSSION THREAD

**Discussion by elpmid** [Blazebit/blaze-persistence#1951] [Q&A] [answered]
Hi - is it possible to enforce join precedence?
In SQL I can use brackets to enforce join precedence, but can't figure out how to do this using the CriteriaBuilder.

SQL Example:
SELECT * FROM table1
 LEFT OUTER JOIN (
       table2 INNER JOIN table3 ON table2.table3_Id = table3.id
      AND table3.column1 = 'SOME_VALUE'
) ON table1.id = table2.table1_Id

In the above the INNER JOIN / filter is done before the LEFT JOIN 

thanks

**Comment by elpmid** [comment]
Can do it with A CTE, but wondering it there is away to do it without

**Comment by beikov** [comment, marked as answer]
There is no definitive way to do this since JPQL/HQL also do not support this.
